internal/bgp/frr/debounce_test.go

```diff
@@ -44,3 +48,69 @@ func TestDebounce(t *testing.T) {
 		t.Fatal("Config was not updated")
 	}
 }
+
+func TestDebounceRetry(t *testing.T) {
+	result := make(chan *frrConfig, 10) // buffered to accomodate spurious rewrites
+	count := 0
+	dummyUpdate := func(config *frrConfig) error {
+		count++
+		if count <= 3 {
+			return fmt.Errorf("error")
+		}
+		result <- config
+		return nil
+	}
+
+	reload := make(chan reloadEvent)
+	defer close(reload)
+	debouncer(dummyUpdate, reload, timer, failureTimer)
+
+	reload <- reloadEvent{config: &frrConfig{Hostname: "1"}}
+	reload <- reloadEvent{config: &frrConfig{Hostname: "2"}}
+	if len(result) != 0 {
+		t.Fatal("received update before time")
+	}
+	time.Sleep(10 * failureTimer)
+	if len(result) != 1 {
+		t.Fatal("received extra updates", len(result))
+	}
+	updated := <-result
+	if updated.Hostname != "2" {
+		t.Fatal("Config was not updated")
+	}
+}
+
+func TestDebounceReuseOld(t *testing.T) {
+	result := make(chan *frrConfig, 10) // buffered to accomodate spurious rewrites
+	dummyUpdate := func(config *frrConfig) error {
+		result <- config
+		return nil
+	}
+
+	reload := make(chan reloadEvent)
+	defer close(reload)
+	debouncer(dummyUpdate, reload, timer, failureTimer)
+
+	reload <- reloadEvent{config: &frrConfig{Hostname: "1"}}
+	if len(result) != 0 {
+		t.Fatal("received update before time")
+	}
+	time.Sleep(3 * timer)
+	if len(result) != 1 {
+		t.Fatal("received extra updates", len(result))
+	}
+	updated := <-result
+	if updated.Hostname != "1" {
+		t.Fatal("Config was not updated")
+	}
+	// reload to see if the debouncer uses the old config
+	reload <- reloadEvent{useOld: true}
+	time.Sleep(3 * timer)
+	if len(result) != 1 {
+		t.Fatal("received extra updates", len(result))
+	}
+	updated = <-result
+	if updated.Hostname != "1" {
+		t.Fatal("Config was not updated")
```

Comment:
shouldn't this be "Config was updated"?

Reply:
Nope, because what writes to result is the action to update the config. So in this case, if we don't get the value, it means that the retry mechanism did not trigger and that the update function was not called